parsers/ECO2MIX.py

```diff
@@ -0,0 +1,295 @@
+    return query(
+        url_type_arg="region", session=session, target_datetime=target_datetime
+    )
+
+
+def parse_production_to_df(text):
+    bs_content = BeautifulSoup(text, features="xml")
+    # Flatten
+    df = pd.DataFrame(
+        [
+            {
+                **valeur.attrs,
+                **valeur.parent.attrs,
+                **valeur.parent.parent.attrs,
+                "value": valeur.contents[0],
+            }
+            for valeur in bs_content.find_all("valeur")
+        ]
+    )
+    if df.empty:
+        return df
+    # Add datetime
+    df["datetime"] = pd.to_datetime(df["date"]) + pd.to_timedelta(
+        df["periode"].astype("int") * 15, unit="minute"
+    )
+    df.datetime = df.datetime.dt.tz_localize("Europe/Paris")
+    # Set index
+    df = df.set_index("datetime").drop(["date", "periode"], axis=1)
+    # Remove invalid granularities
+    df = df[df.granularite == "Global"].drop("granularite", axis=1)
+    # Create key (will crash if a mode is not in the map and ensures we coded this right)
+    df["key"] = df.v.apply(lambda k: MAP_MODES[k])
+    # Filter out invalid modes
+    df = df[df.key != "IGNORED"]
+    # Compute zone_key
+    df["zone_key"] = df["perimetre"].apply(lambda k: MAP_ZONES[k])
+    # Compute values
+    df.value = df.value.replace("ND", np.nan).replace("-", np.nan).astype("float")
+    # Storage works the other way around (RTE treats storage as production)
+    df.loc[df.key.str.startswith("storage."), "value"] *= -1
+    return df
+
+
+def format_production_df(df, zone_key):
+    if df.empty:
+        return []
+    # There can be multiple rows with the same key
+    # (e.g. multiple things lumping into `unknown`)
+    # so we need to group them and sum.
+    df = (
+        df[df.zone_key == zone_key]
+        .reset_index()
+        .groupby(["datetime", "key"])
+        # We use `min_count=1` to make sure at least one non-NaN
+        # value is present to compute a sum.
+        .sum(min_count=1)
+        # We unstack `key` which creates a df where keys are columns
+        .unstack("key")["value"]
+    )
+    return [
+        {
+            "zoneKey": zone_key,
+            "datetime": ts.to_pydatetime(),
+            "production": (
+                row.filter(like="production.")
+                .rename(lambda c: c.replace("production.", ""))
+                .to_dict()
+            ),
+            "storage": (
+                row.filter(like="storage.")
+                .rename(lambda c: c.replace("storage.", ""))
+                .to_dict()
+            ),
+            "source": "rte-france.com/eco2mix",
+        }
+        for (ts, row) in df.iterrows()
+    ]
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_production(
+    zone_key, session=None, target_datetime=None, logger: Logger = getLogger(__name__)
+):
+    if zone_key == "FR-COR":
+        raise ParserException("FR-COR is not supported in this parser")
```

Comment:
```suggestion
        raise ParserException("ECO2MIX.py", "FR-COR is not supported in this parser")
```
ParserException requires a parser values as well.